code/modules/spells/spell_code.dm

```diff
@@ -116,7 +124,51 @@ var/list/spells = typesof(/spell) //needed for the badmin verb for now
 			cast(targets, user)
 		after_cast(targets) //generates the sparks, smoke, target messages etc.
 
+//This is used with the wait_for_click spell flag to prepare spells to be cast on your next click
+/spell/proc/channel_spell(mob/user = usr, skipcharge = 0, force_remove = 0)
+	if(!holder)
+		holder = user //just in case
+	if(!user.spell_channeling && !force_remove)
+		if(!cast_check(skipcharge, user))
+			return 0
+		user.spell_channeling = user.on_uattack.Add(src, "channeled_spell")
+		connected_button.name = "(Ready) [name]"
```

Comment:
The only reason I added in add channeling was to balance the fact I put in remove channeling for GC purposes, otherwise I would have centralized all the code here for easier reading.